subsys/net/lib/coap/Kconfig

```diff
@@ -7,7 +7,7 @@
 #
 
 config COAP
-	bool "CoAP Support"
+	bool "CoAP Support[Deprecated]"
```

Comment:
this is misleading, add more info that only net_pkt mode is being deprecated. I would also have done things differently, keep COAP kconfig and introduce COAP_NET_PKT and COAP_SOCK and avoid duplicating all of the Kconfigs

Reply:
Added a bit more information. I didn't want to modify existing CoAP related stuff. Users of current CoAP can continue as it is except deprecated warnings.

Reply:
> nashif@50bfcb8

Only reason for not doing like that was, you have changed everything in upstream Zephyr who are depending on CoAP. It fails to compile immediately for non upstream. This is the only reason I opted for separate folders and config. I am not arguing about that. When someone has to touch immediately their code after your proposal means why not keep only one implementation in one single go. For reviewers also it is bit easy to see only socket related changes.